Edits to Git entries (11 total) (#562)
Codecademy/docs: content/git/concepts/branch/branch.md

As long as the branch has no uncommitted code, this command will remove the branch entirely. It will no longer be available to check out, and the changed code will be unrecoverable.

Sometimes, Git will refuse to delete a branch because it has unmerged changes in an existing upstream or remote version of that branch. In such cases, it can be forcefully deleted with either the `-d --force` or capitalized `-D` flag, like so in the `my-branch` examples below.

Option 1:

```shell
git branch -d --force my-branch
```

Option 2:

```shell
git branch -D my-branch
```

Performing a force delete tells Git the user understands that all changes on that branch will be lost forever, and they want to delete the branch anyway.

> 

Reviewer: Could we also add a short bit about removing the remote branch equivalent (if one exists)?

```
git push <name-of-remote> --delete <name-branch>
```

Not sure if it should go here b/c it deals with branching or somewhere in `push.md` since this uses the `push` command.

Author: Holding off on this one for now. In the future, it might be worth adding, perhaps side by side with the way to do it on GitHub.

> By performing a force delete, the user is telling Git that they understand all changes on that branch will be lost forever, and they want to perform the delete anyway.

Reviewer 2: I see your point. Alright, then let's stick to the strict terminologies that Git's docs use 👍

As long as the branch has no uncommitted code, this command will remove the branch entirely. It will no longer be available to check out, and the changed code will be unrecoverable.

> Sometimes Git will refuse to delete a branch because it has changes that have not been merged into the upstream or remote version of that branch. In these cases, Git can be _forced_ to delete the branch by changing the flag to a capital "D" or adding the `--force` flag like so:

Reviewer: ```suggestion
Sometimes, Git will refuse to delete a branch because it has changes that have not yet been merged into an existing upstream or remote version of that branch. In such cases, the branch can be forcefully deleted with either the `-d --force` or capital `-D` flags, like so: 
```

Our colleagues in Product prefer to be as minimal with emphasis as possible, which includes leaving out italics in almost all cases.

> ```shell

Reviewer: ```suggestion
```pseudo
```

Reviewer: Since we're using placeholders for real values like `<branch-name>`, let's use ```pseudo for those cases.

> # Option 1:
> git branch -D <branch-name>
> 
> # Option 2:
> git branch -d --force <branch-name>

Reviewer: ```suggestion
# Option 1:
$ git branch -D my-branch

# Option 2:
$ git branch -d --force my-branch
```

After the syntax of a command is introduced, instead of using `<placeholder-syntax>` let's switch to using fictional names like `my-branch`.

Author: I don't know about having the `$` in front of the comments. Those commented lines are not commands that should be entered into the command line. If anything, maybe it is better that they get moved out of the code snippet altogether if each line has to be preceded by `$`

```

By performing a force delete, the user is telling Git that they understand all changes on that branch will be lost forever, and they want to perform the delete anyway.

## Moving Between Branches

> The basic syntax for moving between branches in Git is this:

Reviewer: A couple a thoughts to note about lines #53-57:

* Let's change the sentence in line #53 to exclude "basic" from it.
* On lines #55-57, let's change `<branch-name>` to `branch-name` to make the pseudocode syntactically compliant w/ Git (running with `<`,`>` yields an error).
* * Since this section talks about moving between branches, should we include information about [the `git switch` command](ff3e4e1bce8bcc8068623639aa5f0272e6e6c49f)?

Author: Sure, no problem.

Regarding `git switch`, that might be better added as a separate entry in a new PR, if at all. `git switch` and `git restore` are still considered experimental and I don't personally know anyone that uses them, but they have been around for a while now, so maybe it makes sense to add them.  If nothing else, it may help clear up some confusion about how to use `checkout`.

Reviewer: That sounds good! We should consider those in the near future but not for this PR. 😄
